splunk_topology/test_splunk_topology.py

```diff
@@ -1084,3 +1084,148 @@ def _mocked_auth_session_to_check_instance_config(instance):
         })
         instances = self.check.get_topology_instances()
         self.assertEqual(len(instances), 1)
+
+
+class TestSplunkContinue(AgentCheckTest):
+    CHECK_NAME = 'splunk_topology'
+
+    def test_check_exception_continue(self):
+        """
+        When 1 saved search fails with Check Exception, the code should continue and send topology.
+        """
+        config = {
+            'init_config': {},
+            'instances': [
+                {
+                    'url': 'http://localhost:8089',
+                    'username': "admin",
+                    'password': "admin",
+                    'ignore_saved_search_errors': True,
+                    'component_saved_searches': [
+                        {"name": "components", "search_max_retry_count": 0, "parameters": {}},
+                        {"name": "components12", "search_max_retry_count": 0, "parameters": {}}
+                    ],
+                    'relation_saved_searches': []
+                }
+            ]
+        }
+
+        def _mocked_check_exception_search(*args, **kwargs):
+            sid = args[0]
+            if sid == "components12":
+                raise CheckException("maximum retries reached for saved search " + str(sid))
+            return [json.loads(Fixtures.read_file("%s.json" % sid, sdk_dir=FIXTURE_DIR))]
+
+        self.run_check(config, mocks={
+            '_dispatch_saved_search': _mocked_dispatch_saved_search,
+            '_search': _mocked_check_exception_search,
+            '_saved_searches': _mocked_saved_searches,
+            '_auth_session': _mocked_auth_session
+        })
+        instances = self.check.get_topology_instances()
+        self.assertEqual(len(instances), 1)
+        # Even second saved search failed but topology reported from first saved search
+        self.assertEqual(len(instances[0]['components']), 2)
```

Comment:
can we make sure that the components we get have the expected external id coming from the saved search `components`